specification/billing/resource-manager/Microsoft.Billing/preview/2024-08-02-preview/migration.json

```diff
@@ -0,0 +1,203 @@
+            "$ref": "./types.json#/parameters/billingAccountName"
+          },
+          {
+            "$ref": "./types.json#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Migration mapping details.",
+            "schema": {
+              "$ref": "#/definitions/MigrationMappingResponse"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get migration mapping details": {
+            "$ref": "./examples/GetMigrationMapping.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "Migration"
+        ],
+        "description": "Migrates from source to destination billing scope",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/en-us/rest/api/billing/"
+        },
+        "operationId": "Migration_Put",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "./types.json#/parameters/billingAccountName"
+          },
+          {
+            "$ref": ".,/types.json#/parameters/apiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "parameters",
+            "description": "Migration mapping details.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/MigrationMappingProperties"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Migration details.",
+            "schema": {
+              "$ref": "#/definitions/MigrationResponse"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "CreateOrUpdateMigrationMapping": {
+            "$ref": "./examples/CreateOrUpdateMigrationMapping.json"
+          },
+          "Migration Initiation Details": {
+            "$ref": "./examples/InitiateMigration.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "MigrationMappingProperties": {
+      "description": "Migration mapping properties.",
+      "type": "object",
+      "properties": {
+        "destinationBillingScope": {
+          "description": "Destination billing scope to be mapped.",
+          "type": "string"
+        },
+        "status": {
+          "description": "Migration Request status.",
+          "type": "string"
+        }
+      }
+    },
+    "MigrationResponse": {
```

Comment:
We shouldn't need this. The only definition we need is a "Migration" definition which represents the Migration resource (deriving from e.g. ProxyResource). Then you can use that for both your request and response.